Source/CFRuntime.c

```diff
@@ -384,6 +384,25 @@ CFRetain (CFTypeRef cf)
   return cf;
 }
 
+CFTypeRef
+CFAutorelease (CFTypeRef cf)
+{
+  // FIXME: this is leaking
```

Comment:
Why don't you use objc_autoreleaseReturnValue(cf) from the function below here as well?
You should also be aware that the objc_XXX functions you are using are only present in the libobjc2. You will need extra checks for that.